### ISSUE TITLE
Error installing Zig LSP server: `Error opening archive: Unrecognized archive format`

### ISSUE THREAD

**Issue by hhamud** [doomemacs/doomemacs#7873]
### I confirm that...

- [X] I have searched the [issue tracker](https://github.com/doomemacs/doomemacs/issues), [documentation](https://docs.doomemacs.org), [FAQ](https://docs.doomemacs.org/-/faq), [Discourse](https://discourse.doomemacs.org), and [Google](https://google.com), in case this issue has already been reported/resolved.
- [X] I have read "[How to Debug Issues](https://doomemacs.org/d/how2debug)", and will use it to provide as much information about this issue as possible.
- [X] The issue can be reproduced on the **latest** available commit of Doom.
- [X] The issue can be reproduced on a stable release of Emacs, such as 27, 28, or 29. *(Unstable versions end in .50, .60, or .9x)*

### Expected behavior

When installing the lsp server via doom emacs it should download and install 

### Current behavior

When doom emacs attempts to install the lsp server for zig
```shell
LSP :: Finished downloading /Users/user/.emacs.d/.local/etc/lsp/zig/temp.tar.gz...
LSP :: Decompressing /Users/user/.emacs.d/.local/etc/lsp/zig/temp.tar.gz...
tar: Error opening archive: Unrecognized archive format
LSP :: Decompressed /Users/user/.emacs.d/.local/etc/lsp/zig/temp...
LSP :: Server zls downloaded, auto-starting in 1 buffers.
```

### Steps to reproduce

1. Open Emacs
2. Go to init.el
3. activate zig +lsp
4. Go to any zig folder
5. Activate lsp

### System Information

https://pastebin.com/DVXd5SP1

It is related to this 
https://github.com/emacs-lsp/lsp-mode/issues/4445

**Comment by gchape** [comment]
@hhamud 
`LSP :: Decompressing /Users/user/.emacs.d/.local/etc/lsp/zig/temp.tar.gz...` worked, it seems that the issue might have been transient or related to how the tar command initially interpreted the archive.

Does LSP features work?

**Comment by hhamud** [comment]
> @hhamud `LSP :: Decompressing /Users/user/.emacs.d/.local/etc/lsp/zig/temp.tar.gz...` worked, it seems that the issue might have been transient or related to how the tar command initially interpreted the archive.
> 
> Does LSP features work?

No it does not work at all.

**Comment by hlissner** [comment]
This is simply [an upstream issue](https://github.com/zigtools/zls/issues/1879). ZLS now serves it packages in `tar.xz` files instead of `tar.gz`. We'll have to wait until they either revert or commit to the change, and for `lsp-mode` to update to reflect that.

In the meantime, here is a potential workaround:

```elisp
(unless (featurep :system 'windows)
  (setq lsp-zig-download-url-format "https://github.com/zigtools/zls/releases/latest/download/zls-%s-%s.tar.xz"))
```

**Comment by hlissner** [comment]
Seems to be taking a while for the response, so I've gone ahead and added the corrected `lsp-zig-download-url-format` in 7bb5df4. I'll revert it if the issue is addressed upstream. 

In any case, thanks for bringing this to my attention!
